src/components/Accounts/Actions/ActionModal.js

```diff
@@ -0,0 +1,509 @@
+    super(props);
+
+    this.state = {
+      amount: '0.00',
+      showNotify: false,
+      notify: false,
+    };
+  }
+
+  shouldComponentUpdate(nextProps, nextState) {
+    const {
+      accounts,
+      feefines,
+      owners,
+      initialize,
+      open,
+      pristine,
+      invalid,
+    } = this.props;
+    if (!_.isEqual(accounts, nextProps.accounts)) {
+      let showNotify = false;
+      let notify = false;
+      const amount = calculateSelectedAmount(nextProps.accounts);
+      nextProps.accounts.forEach(a => {
+        const feefine = feefines.find(f => f.id === a.feeFineId) || {};
+        const owner = owners.find(o => o.id === a.ownerId) || {};
+        if (feefine.actionNoticeId || owner.defaultActionNoticeId) {
+          showNotify = true;
+          notify = true;
+        }
+      });
+      this.setState({
+        amount,
+        showNotify,
+        notify
+      });
+      if (nextProps.form === 'payment-many-modal') {
+        initialize({ amount, notify: true, ownerId: this.loadServicePoints(nextProps) });
+      } else {
+        initialize({ amount, notify: true });
+      }
+    }
+
+    return (accounts !== nextProps.accounts
+      || this.state !== nextState ||
+      open !== nextProps.open ||
+      pristine !== nextProps.pristine ||
+      invalid !== nextProps.invalid);
+  }
+
+  loadServicePoints = (props) => {
+    const servicePoint = props.defaultServicePointId;
+    const servicePoints = props.servicePointsIds;
+    const owners = props.owners || [];
+    let ownerId = 0;
+    if (servicePoint && servicePoint !== '-') {
+      owners.forEach(o => {
+        if (o.servicePointOwner && o.servicePointOwner.find(s => s.value === servicePoint)) {
+          ownerId = o.id;
+          this.onChangeOwner({ target: { value: o.id } });
+        }
+      });
+    } else if (servicePoints.length === 1) {
+      const sp = servicePoints[0];
+      owners.forEach(o => {
+        if (o.servicePointOwner && o.servicePointOwner.find(s => s.value === sp)) {
+          ownerId = o.id;
+          this.onChangeOwner({ target: { value: o.id } });
+        }
+      });
+    } else if (servicePoints.length === 2) {
+      const sp1 = servicePoints[0];
+      const sp2 = servicePoints[1];
+      owners.forEach(o => {
+        if (o.servicePointOwner && o.servicePointOwner.find(s => s.value === sp1) && o.servicePointOwner.find(s => s.value === sp2)) {
+          ownerId = o.id;
+          this.onChangeOwner({ target: { value: o.id } });
+        }
+      });
+    }
+    return ownerId;
+  }
+
+  onChangeAmount = (e) => {
+    this.setState({
+      amount: parseFloat(e.target.value || 0).toFixed(2),
+    });
+  }
+
+  onToggleNotify = () => {
+    this.setState(prevState => ({
+      notify: !prevState.notify,
+    }));
+  }
+
+  onClose = () => {
+    const {
+      accounts,
+      onClose,
+      reset,
+    } = this.props;
+
+    const selected = calculateSelectedAmount(accounts); // Por que?
```

Comment:
`// Por que?`
Please write comments in English. As I see this comment is not informative. If it is not needed at all, please remove it.